fg_lib/include/fg_lib.h

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef __FG_LIB_H__
+#define __FG_LIB_H__
+
+/**
+ * @brief Algorithm for battery state-of-charge estimation ("fuel gauge").
+ *
+ * @details Algorithm requires a model of the specific battery model used in the given application.
+ *          This model, along with measurements of battery voltage, current, and temperature,
+ *          will determine the predicted state-of-charge.
+ *          The state-of-charge, or "SOC", is a percentage value between 0% and 100%.
```

Comment:
```suggestion
@details The algorithm determines the predicted state-of-charge (SOC) based on the following data: model of the specific battery type used in the given application along with measurements of battery voltage, current, and temperature.
The SOC is a percentage value between 0% and 100%.
```
A model of a model? Can we say "battery type"? Also because the next sentence starts with "this model", but so it could point to two different instances of the model word.